### PR TITLE
Adding Icosahedral Sphere

### PR DESCRIPTION
**Pull request by Rhuagh**

### CODE REVIEW

File: src/icosphere.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
+//   Copyright GFX Developers 2014-2017
```
Comment (kvark) [inline src/icosphere.rs]: a separate PR needs to clean out all the licensing from files

File: src/icosphere.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+/// Icosahedral sphere with radius 1, centered at (0., 0., 0.)
+#[derive(Clone, Debug)]
+pub struct IcoSphere {
+    i: usize,
+    vertices: Vec<[f32; 3]>,
+    faces: Vec<[usize; 3]>,
+}
+
+const T: f32 = 0.85065080835204;
```
Comment (kvark) [inline src/icosphere.rs]: where do these come from?

Comment (Rhuagh) [reply src/icosphere.rs]: They come from calculating the sides of a rectangle inside the sphere.

Comment (Rhuagh) [reply src/icosphere.rs]: An icosahedrons vertices can be defined as the corner points of three orthogonal golden rectangles, which means the sides of the rectangles have the golden ratio: (1 + sqrt(5)) / 2. If you take that directly though you won't get a unit sphere, so we need to normalize the points (0, ±1, ±ratio), which gives those 2 values. I precomputed those so it wouldn't be necessary runtime.

**Comment by kvark** [comment]
Thanks! That should be in comments

**Pull request merged**